
<div class="container">
    <div class="row">
        <div class="col-lg-12">
            <div class="modal-body">
                <h2>Microservices with Load Balancing</h2>
                <hr class="star-primary">
                <img src="images/portfolio/Microservices.png" class="img-responsive img-centered" alt="">
                <p>In this example I have 5 services written in different technologies (Java, C#, F#, Javascript, Python). 
                    Each microservices has the same contract are all of them are under the same load balancer. </p>
                <p>The angular application is making a call to the loadbalancer, binding to the view and showing the result.</p>
                <form ng-submit="submit()" class="form-inline">
                    <div class="form-group">
                        <label class="sr-only" for="number1">Number 1</label>
                        <input type="text" class="form-control" id="number1" placeholder="Number 1" ng-model="num1">
                    </div>
                    +
                    <div class="form-group">
                        <label class="sr-only" for="number2">Number 2</label>
                        <input type="text" class="form-control" id="number2" placeholder="Number 2" ng-model="num2">
                    </div>
                    <button type="submit" class="btn btn-primary">Calculate</button>
                </form>
                <p>
                <div class="row">
                    <div class="col-md-3 col-md-offset-4 result" ng-model="result">Result : {{ result }}</div>
                    <div class="col-md-3 col-md-offset-4 from" ng-model="from">From : {{ from }}</div>
                </div>
                </p>
                <h4>Let's play with it: </h4>
                <p>The Load balancer NGINX will retry a failing request to any server.</p>
                <p>To test this will be shuting down one of the instances and see how the request never is missing.</p>
                <p>Go to docker terminal and run "docker stop fullstacj_node3_1"</p>
                <p>And try again but this time look at the request. You wont find any 500 errors in the requests</p>
                <p>Let's check a bit closer. Let's see the logs in the load balancer with "docker logs  fullstacj_nginx_1".</p>
                <p> You can find that when a request fail, it triggers automatically another one.</p>                                                       
            </div>
        </div>
    </div>
</div>
